diff --git a/utils/utils_lilv.cpp b/utils/utils_lilv.cpp
index 4ac28101..19555883 100644
--- a/utils/utils_lilv.cpp
+++ b/utils/utils_lilv.cpp
@@ -4129,7 +4129,7 @@ const PedalboardInfo_Mini* const* get_all_pedalboards(void)
 
     // Custom path for pedalboards
     const char* const oldlv2path = getenv("LV2_PATH");
-    setenv("LV2_PATH", "~/.pedalboards/", 1);
+    setenv("LV2_PATH", "/mnt/pedalboards/:~/.pedalboards/", 1);
 
     LilvWorld* const w = lilv_world_new();
     lilv_world_load_all(w);
@@ -4190,7 +4190,7 @@ const char* const* get_broken_pedalboards(void)
 
     // Custom path for pedalboards
     const char* const oldlv2path = getenv("LV2_PATH");
-    setenv("LV2_PATH", "~/.pedalboards/", 1);
+    setenv("LV2_PATH", "/mnt/pedalboards/:~/.pedalboards/", 1);
 
     LilvWorld* const w = lilv_world_new();
     lilv_world_load_all(w);
