reverted:
--- b/Makefile
+++ a/Makefile
@@ -4,12 +4,12 @@
 VERSION=0.9.71
 
 CC?=gcc
+CXX?=g++
-CPP=g++
 
 OPTIMIZE=-O3
 #OPTIMIZE=-O0 -g
 
+COMPILEFLAGS=$(CFLAGS) $(OPTIMIZE) -DVERSION=\"$(VERSION)\" -Wall -Wextra -Wno-unused -Wno-unused-parameter
-COMPILEFLAGS=$(CFLAGS) $(OPTIMIZE) -DVERSION=\"$(VERSION)\" -Wall -Wextra -Wno-unused
 LINKFLAGS=$(LDFLAGS) -ljack -lsndfile -lm -lpthread -lrt
 
 targets = jack_capture
@@ -33,7 +33,7 @@
 	which install
 	which pkg-config
 	which $(CC)
+	which $(CXX)
-	which $(CPP)
 	$(CC) -E testsndfile.c >/dev/null
 	@echo "All seems good "
 	@echo
@@ -56,7 +56,7 @@
 
 
 jack_capture_gui2: jack_capture_gui2.cpp
+	$(CXX) $(CXXFLAGS) $(OPTIMIZE) jack_capture_gui2.cpp $(LDFLAGS) `pkg-config --libs --cflags gtk+-2.0` -o jack_capture_gui2
-	$(CPP) $(CPPFLAGS) $(OPTIMIZE) jack_capture_gui2.cpp $(LDFLAGS) `pkg-config --libs --cflags gtk+-2.0` -o jack_capture_gui2
 
 config_flags: Makefile das_config.h
 	cat das_config.h |grep COMPILEFLAGS|sed s/\\/\\/COMPILEFLAGS//|tr '\n' ' ' >config_flags
reverted:
--- b/gen_das_config_h.sh
+++ a/gen_das_config_h.sh
@@ -16,7 +16,7 @@
 echo "#include <sndfile.h>" >temp$$.c
 echo "main(){return SF_FORMAT_OGG;}" >>temp$$.c
 echo >>temp$$.c
+if $CC $CFLAGS $LDFLAGS temp$$.c 2>/dev/null; then
-if gcc temp$$.c 2>/dev/null; then
     echo "#define HAVE_OGG 1"
 else
     echo "#define HAVE_OGG 0"
@@ -26,7 +26,7 @@
 echo "#include <lame/lame.h>" >temp$$.c
 echo "main(){return 0;}" >>temp$$.c
 echo >>temp$$.c
+if $CC $CFLAGS $LDFLAGS temp$$.c -lmp3lame 2>/dev/null; then
-if gcc temp$$.c -lmp3lame 2>/dev/null; then
     echo "#define HAVE_LAME 1"
     echo "//COMPILEFLAGS -lmp3lame"
 else
@@ -37,7 +37,7 @@
 echo "#include <lo/lo.h>" >temp$$.c
 echo "main(){return 0;}" >>temp$$.c
 echo >>temp$$.c
+if pkg-config --cflags --libs liblo >/dev/null 2>/dev/null && $CC $CFLAGS $LDFLAGS temp$$.c `pkg-config --cflags --libs liblo` 2>/dev/null ; then
-if pkg-config --cflags --libs liblo >/dev/null 2>/dev/null && gcc temp$$.c `pkg-config --cflags --libs liblo` 2>/dev/null ; then
     echo "#define HAVE_LIBLO 1"
     echo "//COMPILEFLAGS " `pkg-config --cflags --libs liblo`
 else
@@ -48,7 +48,7 @@
 echo "#include <jack/jack.h>" >temp$$.c
 echo "main(){return (int)jack_port_get_latency_range;}" >>temp$$.c
 echo >>temp$$.c
+if $CC $CFLAGS $LDFLAGS temp$$.c -ljack 2>/dev/null ; then
-if gcc temp$$.c -ljack 2>/dev/null ; then
     echo "#define NEW_JACK_LATENCY_API 1"
 else
     echo "#define NEW_JACK_LATENCY_API 0"
reverted:
--- b/gen_setformat_c.sh
+++ a/gen_setformat_c.sh
@@ -57,7 +57,7 @@
     echo "#include <sndfile.h>" >temp.c
     echo "main(){return SF_FORMAT_"$a";}" >>temp.c
     echo >>temp.c
+    if $CC $CFLAGS $LDFLAGS temp.c 2>/dev/null; then
-    if gcc temp.c 2>/dev/null; then
 	echo "    (!strcasecmp(\""$a"\",soundfile_format)) ? SF_FORMAT_"$a":"
     fi
 done
